### ISSUE TITLE
negative loss

### ISSUE THREAD

**Issue by marooncn** [keras-team/keras#12423]
Hi, I define the loss as -log(P(y_true)), P~N(mu, sigma). The code is 
```
 def rnn_r_loss(y_true, y_pred):
     result = y_true - mu
     result = result * (1.0 / (sigma + 1e-8))
     result = -K.square(result) / 2.0
     result = K.exp(result) * 1.0 / ((math.sqrt(2 * math.pi))*(sigma + 1e-8))
     loss= -K.log(result + 1e-8)
     loss = K.mean(loss, axis=1)  
     return loss
```
But when I train the model, the loss decreases to negative value (-1.7) . But as 0 < P(y_true) <= 1, in theory, it can't be negative. How does it happen and how can I solve the problem? Thank you.

**Comment by adrian-spataru** [comment]
I have a similar loss function and I'm experiencing the same issue. My loss function, after several epochs, becomes negative as well. The reason that your loss is negative is because `result` is bigger than 1. (log(1.something) is positive) Furthermore, I think it makes more sense to do something like `-K.log(tf.maximum(result, 1e-8))`, but this is surely not what causes the issue. I would also appreciate some thoughts on this.

**Comment by marooncn** [comment]
@adrian-spataru , yes, but theoretically **result** is the probability under Gaussian distribution, it can't be bigger than 1. I wonder if it's because the inner mechanism in Keras.

**Comment by adrian-spataru** [comment]
@marooncn True. I assume mu and sigma is calculated from a Dense Layer, where sigma  is calculated from `sigma = exponential(sigma)` or similar. What might have fixed the issue for me, is to put a sigmoid activation to that Dense layer and have the values scaled betwen 0 and 1. Even though theoretically it should make no difference.

**Comment by marooncn** [comment]
@adrian-spataru, Hi, that's so amazing. I add **K.sigmoid(sigma)** and then train it and the training process is normal now:
![2019-03-09 16-22-00屏幕截图](https://user-images.githubusercontent.com/29606478/54068421-99364680-4287-11e9-8723-31b3e418c6a4.jpg)
That's so weird, why does it? And sigma can only betwen 0 and 1 now. I think it's a Keras bug. Hope to solve it.

**Comment by adrian-spataru** [comment]
@marooncn Awesome! As long your input is also scaled between 0 and 1 then it should run fine. Is it due to the fact and sigma must be positive? As the standard deviation is the square root of the variance?

**Comment by marooncn** [comment]
@bmabey @smgt @moustaki , hope to explain it. And what if sigma is more than 1?
